File: instedit/_1st_marked.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

// Returns index of instrument + 1 (1..255) if found, 0 otherwise.
uint8_t _1st_marked (void)
{
  for (uint_least8_t i = 0; i <= 254; i++)
    if (GetStr (songdata.instr_names[i])[0] == _INS_MARK_CHR)
      return i + 1;

  return 0;
}

File: instedit/pas/_1st_marked.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

function _1st_marked: Byte;

var
  temp,result: Byte;

begin
  result := 0;
  For temp := 1 to 255 do
    If (songdata.instr_names[temp][1] = _ins_mark_chr) then
      begin
        result := temp;
        BREAK;
      end;
  _1st_marked := result;
end;
